homeassistant/components/zha/sensor.py

```diff
@@ -56,279 +141,61 @@
-                'battery_quantity',
-                'battery_percentage_remaining'
-            ],
-            allow_cache=False,
-            only_cache=(not self._initialized)
+        self._force_update = FORCE_UPDATE_REGISTRY.get(
+            sensor_type,
+            False
+        )
+        self._should_poll = POLLING_REGISTRY.get(
+            sensor_type,
+            False
+        )
+        self._listener = self.get_listener(
+            LISTENER_REGISTRY.get(sensor_type, LISTENER_ATTRIBUTE)
         )
-        self._device_state_attributes['battery_size'] = self.battery_sizes.get(
-            result.get('battery_size', 255), 'Unknown')
-        self._device_state_attributes['battery_quantity'] = result.get(
-            'battery_quantity', 'Unknown')
-        self._state = result.get('battery_percentage_remaining', self._state)
-
-    @property
-    def state(self):
-        """Return the state of the entity."""
-        if self._state == 'unknown' or self._state is None:
-            return None
-
-        return self._state
-
-
-class TemperatureSensor(Sensor):
-    """ZHA temperature sensor."""
-
-    min_reportable_change = 50  # 0.5'C
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement of this entity."""
-        return self.hass.config.units.temperature_unit
-
-    @property
-    def state(self):
-        """Return the state of the entity."""
-        if self._state is None:
-            return None
-        celsius = self._state / 100
-        return round(convert_temperature(celsius,
-                                         TEMP_CELSIUS,
-                                         self.unit_of_measurement),
-                     1)
-
-
-class RelativeHumiditySensor(Sensor):
-    """ZHA relative humidity sensor."""
-
-    min_reportable_change = 50  # 0.5%
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement of this entity."""
-        return '%'
-
-    @property
-    def state(self):
-        """Return the state of the entity."""
-        if self._state is None:
-            return None
-
-        return round(float(self._state) / 100, 1)
-
 
-class PressureSensor(Sensor):
-    """ZHA pressure sensor."""
+    async def async_added_to_hass(self):
+        """Run when about to be added to hass."""
+        await super().async_added_to_hass()
+        await self.async_accept_signal(
+            self._listener, SIGNAL_ATTR_UPDATED, self.set_state)
+        await self.async_accept_signal(
+            self._listener, SIGNAL_STATE_ATTR, self.update_state_attribute)
 
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity."""
-        return 'hPa'
+        return self._unit
 
     @property
-    def state(self):
+    def state(self) -> str:
         """Return the state of the entity."""
-        if self._state is None:
+        if self._state is STATE_UNKNOWN or self._state is None:
```

Comment:
When would `_state` be `STATE_UNKNOWN`. We shouldn't use that internally. Use `None` to represent unknown state. Are we restoring state anywhere?

Reply:
I’ll take a look at why this is like this.

Reply:
gone